http.go

```diff
@@ -43,13 +40,38 @@ const (
 
 // GetHTTPClient returns either a standard http client or
 // non validating client depending on the value of verify.
-func GetHTTPClient(verify SSLHostnameVerification) *http.Client {
+func GetHTTPClient(verify SSLHostnameVerification, certs ...string) *http.Client {
+	if len(certs) > 0 {
+		return GetHTTPClientWithCerts(verify, certs)
+	}
 	if verify == VerifySSLHostnames {
 		return GetValidatingHTTPClient()
 	}
 	return GetNonValidatingHTTPClient()
 }
 
+// GetHTTPClientWithCerts returns a new http.Client that verifies the
+// server's certificate chain and hostname depending on arguments and
+// adds ca certificates to the client. Returns nil if no certificates
+// provided.
+func GetHTTPClientWithCerts(verify SSLHostnameVerification, certs []string) *http.Client {
```

Comment:
Given that the `GetHTTPClient` signature is essentially the same, would it be worth making this private so we can force package importers to go through `GetHTTPClient`?

Reply:
done